ding_periodical.module

```diff
@@ -135,12 +136,17 @@ function theme_ding_periodical_issues($variables) {
 
       // Check if the volume is reservable and add reservation button.
       if (!empty($issue_id) && ding_periodical_is_reservable($issues[$key])) {
-        $path = array($issue_id, $volume, $key);
+        $item_id = array($issue_id, $volume, $key);
+
+        foreach ($item_id as &$v) {
+          $v = rawurlencode($v);
```

Comment:
OK. I agree that if other providers used the same form of periodical ids then they would have the same problem. This should fix it once and for all.

As I reread the code I have another question regarding the `rawurlencode()`: Why is this necessary?

I would think that the subsequent call to `base64_encode()` eliminated problems with programmatic characters. There could be an odd chance that the value of `$issue_id`, `$volume` or `$key` contained a `|` but I would think that could be handled by using more complex delimiter for implode/explode.